docs/proposals/templates.md

```diff
@@ -0,0 +1,595 @@
+          }
+        ],
+        "selector": {
+          "name": "$(DATABASE_SERVICE_NAME)"
+        }
+      }
+    },
+    {
+      "kind": "ReplicationController",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "$(DATABASE_SERVICE_NAME)",
+      },
+      "spec": {
+        "replicas": "$((REPLICA_COUNT))",
+        "selector": {
+          "name": "$(DATABASE_SERVICE_NAME)"
+        },
+        "template": {
+          "metadata": {
+              "creationTimestamp": null,
+              "labels": {
+                  "name": "$(DATABASE_SERVICE_NAME)"
+              }
+          },
+          "spec": {
+            "containers": [
+              {
+                "name": "mongodb",
+                "image": "docker.io/centos/mongodb-26-centos7",
+                "ports": [
+                  {
+                    "containerPort": 27017,
+                    "protocol": "TCP"
+                  }
+                ],
+                "env": [
+                  {
+                    "name": "MONGODB_USER",
+                    "value": "${MONGODB_USER}"
+                  },
+                  {
+                    "name": "MONGODB_PASSWORD",
+                    "value": "${MONGODB_PASSWORD}"
+                  },
+                  {
+                    "name": "MONGODB_DATABASE",
+                    "value": "${MONGODB_DATABASE}"
+                  }
+                ]
+              }
+            ]
+          }
+        }
+      }
+    }
+  ],
+  "parameters": [
+    {
+      "name": "DATABASE_SERVICE_NAME",
+      "description": "Database service name",
+      "value": "mongodb",
+      "required": true
+    },
+    {
+      "name": "MONGODB_USER",
+      "description": "Username for MongoDB user that will be used for accessing the database",
+      "value": "password"
+      "required": true
+    },
+    {
+      "name": "MONGODB_PASSWORD",
+      "description": "Password for the MongoDB user",
+      "required": true
+    },
+    {
+      "name": "MONGODB_DATABASE",
+      "description": "Database name",
+      "value": "sampledb",
+      "required": true
+    },
+    {
+      "name": "REPLICA_COUNT",
+      "description": "Number of mongo replicas to run",
+      "value": "1",
+      "required": true
+    }
+  ]
+}
+```
+
+### API Endpoints
+
+* **/processedTemplates** - when a template is POSTed to this endpoint, all parameters in the template are processed, values 
+generated as necessary, and substituted into appropriate locations in the object definitions.  Validation is performed to ensure
+required parameters have a value (either supplied in the original template or generated by a defined generator).  In addition 
+labels defined in the template are applied to the object definitions.  Finally the customized template (still a `Template` object) 
+is returned to the caller.  (Would we prefer to return an ObjectList instead of a Template object, to make it easier for clients
```

Comment:
I guess returning a Template is more consistent with other endpoints, but is a bit more work for clients. I'm fine with the former for now.